src/ray/gcs/gcs_server/gcs_resource_manager.h

```diff
@@ -88,6 +78,13 @@ class GcsResourceManager : public rpc::NodeResourceInfoHandler {
   const absl::flat_hash_map<NodeID, std::shared_ptr<SchedulingResources>>
       &GetClusterResources() const;
 
+  /// Update resources of a node
+  void UpdateResources(const NodeID &node_id,
+                       absl::flat_hash_map<std::string, double> changed_resources);
+
+  /// Delete resource of a node
+  void DeleteResources(const NodeID &node_id, std::vector<std::string> resource_names);
```

Comment:
Same questino as above 

```suggestion
  void DeleteResources(const NodeID &node_id, std::vector<std::string> &resource_names);
```

Reply:
@rkooo567  The non-const is on purpose because we need to copy the struct anyway (please check the implementation).
In this way, the caller can move the object to avoid copy.